main.py

```diff
@@ -11,39 +11,47 @@
 from tornado.web import asynchronous
 import json
 import random
+import uuid
+
 
 # Global ID seed for clients
 clients_connected = 0
-proxy = None
-clients = []
-connToClient = None
+proxies = []
+clients = {}
 
 
 class HttpHandler(tornado.web.RequestHandler):
     @asynchronous
     def get(self):
-        global proxy, connToClient
+        global proxies, clients
 
-        print "got get"
         args = {}
         args['topic'] = self.get_argument('topic')
         args['height'] = self.get_argument('height','480')
         args['width'] = self.get_argument('width','640')
 
-        self.clear()
-        self.set_status(200)
-        self.set_header('server', 'example')
-        self.set_header('connection', 'close')
-        self.set_header('pragma', 'no-cache')
-        self.set_header('cache-control', 'no-cache, no-store, must-revalidate,'
+        session_id = self.get_cookie('session_id')
+        if session_id != None:
+
+            self.set_status(200)
+            self.set_header('server', 'example')
+            self.set_header('connection', 'close')
+            self.set_header('pragma', 'no-cache')
+            self.set_header('cache-control', 'no-cache, no-store, must-revalidate,'
                         'pre-check=0, post-check=0, max-age=0')
-        self.set_header('access-control-allow-origin', '*')
-        self.set_header('content-type', 'multipart/x-mixed-replace;boundary='
+            self.set_header('access-control-allow-origin', '*')
+            self.set_header('content-type', 'multipart/x-mixed-replace;boundary='
                         '--boundarydonotcross')
-        if proxy is not None:
-            connToClient = self
-            message = json.dumps({"op":"video", "args" : args})
-            proxy.write_message(message)
+
+            client = clients.get(session_id)
+            if client != None and client.proxy != None:
+                if client.authenticated or not proxy.user_auth:
+                    message = json.dumps({"op":"videoStart", "url_params" : args, "session_id": session_id})
+                    client.video_conn = self
+                    client.proxy.conn.write_message(message)
+                    return
+        self.set_status(401)
```

Comment:
TODO: include a clear error response which the user can use to figure out why the request failed